influxdb/client.py

```diff
@@ -101,6 +107,70 @@ def __init__(self,
             'Content-type': 'application/json',
             'Accept': 'text/plain'}
 
+    @staticmethod
+    def from_DSN(dsn, **kwargs):
+        """
+        Returns an instance of InfluxDBClient from the provided data source
+        name. Supported schemes are "influxdb", "https+influxdb",
+        "udp+influxdb". Parameters for the InfluxDBClient constructor may be
+        also be passed to this function.
+
+        Examples:
+            >>> cli = InfluxDBClient.from_DSN('influxdb://username:password@\
+            ... localhost:8086/databasename', timeout=5)
+            >>> type(cli)
+            <class 'influxdb.client.InfluxDBClient'>
+            >>> cli = InfluxDBClient.from_DSN('udp+influxdb://username:pass@\
+            ... localhost:8086/databasename', timeout=5, udp_port=159)
+            >>> print('{0._baseurl} - {0.use_udp} {0.udp_port}'.format(cli))
+            http://localhost:8086 - True 159
+
+        :param dsn: data source name
+        :type dsn: string
+        :param **kwargs: additional parameters for InfluxDBClient.
+        :type **kwargs: dict
+        :note: parameters provided in **kwargs may override dsn parameters.
+        :note: when using "udp+influxdb" the specified port (if any) will be
+        used for the TCP connection; specify the udp port with the additional
+        udp_port parameter (cf. examples).
+        :raise ValueError: if the provided DSN has any unexpected value.
+        """
+        dsn = dsn.lower()
+
+        init_args = {}
+        conn_params = urlparse(dsn)
+        scheme_info = conn_params.scheme.split('+')
+        if len(scheme_info) == 1:
+            scheme = scheme_info[0]
+            modifier = None
+        else:
+            modifier, scheme = scheme_info
+
+        if scheme != 'influxdb':
+            raise ValueError('Unknown scheme "{}".'.format(scheme))
+        if modifier:
+            if modifier == 'udp':
+                init_args['use_udp'] = True
+            elif modifier == 'https':
```

Comment:
@gst , specify `https` in the examples, or have the possibility to specify a DSN such as `udp+https+influxdb://` (I don't even know if that's a valid DSN)? 

@ReAzem Effectivement!